luthier/config/src/utils/dimensionUtils.lua

```diff
@@ -52,10 +55,13 @@ function M.build_dimensions_config(dimensions)
     local configuration = {}
     for name, dimension in pairs(dimensions) do
         local dim_copy = misc.shallow_copy(dimension)
-        dim_copy.apiName = dim_copy.apiName or name
         dim_copy.longName = dim_copy.longName or name
         dim_copy.description = dim_copy.description or name
-        configuration[dim_copy.apiName] = dim_copy
+        dim_copy.type = dim_copy.type or default_type
+        dim_copy.category = dim_copy.category or default_category
+        dim_copy.defaultFields = dim_copy.defaultFields or {}
+        if dim_copy.isAggregatable == nil then dim_copy.isAggregatable = true end  -- defaults isAggregatable to true
```

Comment:
Personally, I would wrap the if statement:
```lua
if dim_copy.isAggregatable == nil then
dim_copy.isAggregatable = true
end
```
It's clearer in my opinion: it highlights the important part: the assignment.

Reply:
Sure.